include/boost/range/irange.hpp

```diff
@@ -205,6 +205,22 @@ namespace boost
 
     template<typename Integer>
     integer_range<Integer>
+    irange(int last)
+    {
+        BOOST_ASSERT( 0 <= last );
+        return integer_range<Integer>(0, last);
+    }
+
+    template<typename Integer>
+    integer_range<Integer>
+    irange(int first, int last)
+    {
+        BOOST_ASSERT( first <= last );
+        return integer_range<Integer>(first, last);
+    }
+
```

Comment:
Lines 214-220 aren't actually required. The case when first and last are integers is handled by the method on line 224.